test/tests/utils/test.helpers.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+define(['chai'],(chai)=>{
+    const safeBins = p5.prototype.safeBins;
+    let expect = chai.expect;
+
+    describe('safeBins', ()=>{
+        it('can handle negative input ',()=>{
+            let test =-Math.random()*2;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle value less than 16 ',()=>{
+            let test = 10;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle value greater than 1024',()=>{
+            let test = 1500;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle value other than power 2',()=>{
+            let test = 1500;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle value equal to 0',()=>{
+            let test = 0;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle strings',()=>{
+            let test = 'testString';    
+            expect(safeBins(test)).to.equal(1024);
```

Comment:
all of these tests "handle" _invalid_ values by setting a default value.

Other libraries might choose to throw an error when a user enters an invalid value. Since we're not throwing an error, I wonder if there's a way we could verify that an error message was logged to the console?


It is useful to test behavior when a user provides invalid values. But we probably also want to test for valid value(s). For example, with the tests so far, the method could be changed to _always_ return 1024 and we wouldn't realize this bug.

Reply:
ya ! this is the bug , if anyone changes the function to always return 1024 then our also this test will pass . okay i will be including that